jax/_src/api.py

```diff
@@ -2407,7 +2407,8 @@ def __init__(self, shape, dtype):
   return tree_unflatten(out_tree(), out)
 
 
-def checkpoint(fun: Callable, concrete: bool = False) -> Callable:
+def checkpoint(fun: Callable, concrete: bool = False, prevent_cse: bool = False,
```

Comment:
Do we need to change the API here? How JAX tells XLA to remat should not be a concern for the user. 
If we do need this flag now, perhaps we can name it or document it explicitly as experimental.

Reply:
Ah, you're right that this could be internal only. I forget why I added it to the API; it may be that @jekbradbury suggested adding it from experience on some of his work. But I'll change this so that it's internal-only until we need it!

Reply:
I wanted this so that I could set it to False when using remat inside scan (which already prevents CSE between the forward and backward passes).

Reply:
Ah, then I've done this wrong. I think I know how to fix it.